### PR TITLE
#164797148 Create User Profile feature

### PR DESCRIPTION
**Pull request by eben-k**
#### What does this PR do?
Has user profile update and viewing feature created

#### Description of Task to be completed?
* Write unit tests
*  Create update user profile endpoint

#### How should this be manually tested?
* Clone the repo,
* Run npm install toInstall all dependencies,
* Create .env file using the .env.example as a guide and enter you environment variables
* Run npm test
* Run npm run start:dev,
* On Postman, PUT json/application required inputs to localhost:3000/api/v1/users/profile/:userId

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?(if applicable)
#164797148

#### Screenshots
image

### CODE REVIEW

**Comment by musonant** [comment]
Do you have ESLint and Prettier extensions installed on your text editor? It would help prevent a lot of the issues pointed out by Hound.

I do

File: src/db/models/User.js [hunk 1]
```diff
@@ -42,9 +48,9 @@ module.exports = (sequelize, DataTypes) => {
       beforeUpdate: user => user.password && user.hashPassword()
     },
   });
```
Comment (Japhethca) [inline src/db/models/User.js]: lets remove this comment.

File: src/db/models/User.js [hunk 2]
```diff
@@ -42,9 +48,9 @@ module.exports = (sequelize, DataTypes) => {
...
-  User.associate = () => {
-    // associations can be defined here
-  };
+  // associations can be defined here
```
Comment (Japhethca) [inline src/db/models/User.js]: let's remove this comment.

File: src/tests/db/controller/user.test.js [hunk 3]
```diff
@@ -58,3 +60,43 @@ describe('CREATE USER', () => {
...
+      });
+  });
+});
+describe('Get a single User', () => {
+
+ it('Should not get a non-existent user',
+  async () => {
+  const res = await chai.request(app)
+        .get('/api/v1/user/johndoe');
+      // .set('Authorization', firstUserToken);
```
Comment (Japhethca) [inline src/tests/db/controller/user.test.js]: let's remove this comment.

File: src/tests/db/controller/user.test.js [hunk 4]
```diff
@@ -58,3 +60,43 @@ describe('CREATE USER', () => {
...
+      expect(res.status).to.equal(404);
+      expect(res.body.message)
+        .to.equal('The username provided does not exist');
+    });
+
+  it('Should get a user with valid username param',
+    async () => {
+      const res = await chai.request(app)
+    .get('/api/v1/user/janesmith')
+    // .set('Authorization', firstUserToken);
```
Comment (Japhethca) [inline src/tests/db/controller/user.test.js]: same as above

File: src/db/models/User.js (hunk 2 above)
Comment (eben-k) [reply src/db/models/User.js]: Okay.  Done

File: src/routes/index.js [hunk 5]
```diff
@@ -1,8 +1,11 @@
 import { Router } from 'express';
-import { createUser, confirmUser, } from './controllers/users';
-
+import passport from './services/passport-strategies';
+import { createUser, confirmUser, socialController } from './controllers/users';
+import { forgotPassword, resetPassword } from './controllers/password';
 import validateNewUser from './middlewares/validateUser';
+import userRouter from './users.routes';
 
+// const { verifyToken, getUserId } = Jwt;
```
Comment (Japhethca) [inline src/routes/index.js]: Let's remove this comment here.

Comment (eben-k) [reply src/routes/index.js]: It will be removed ASAP

Comment (eben-k) [reply src/routes/index.js]: Done!

File: src/routes/controllers/users.controller.js [hunk 6]
```diff
@@ -49,3 +54,66 @@ export async function getAuthorProfile(req, res) {
...
 }
+/**
+ * Get current User
+ * @constructor
+ * @param {object} req
+ * @param {object} res
+ * @param {uuid} userId - The uuid variable.
+ */
+export const getUserProfile = async (req, res) => {
+  const { url } = req;
```
Comment (Japhethca) [inline src/routes/controllers/users.controller.js]: we can destructure our variables in one line. 
```javascript 
const {url, user} = req;
```

File: src/routes/controllers/users.controller.js [hunk 7]
```diff
@@ -49,3 +54,66 @@ export async function getAuthorProfile(req, res) {
...
+  const { user } = req;
+  const { userId } = req.params;
+  let userFound;
+
+  try {
+    if (url.includes('profiles')) {
```
Comment (Japhethca) [inline src/routes/controllers/users.controller.js]: We can create a new route/endpoint for this.

File: src/routes/utils/helpers.js [hunk 8]
```diff
@@ -0,0 +1,53 @@
+import jwt from 'jsonwebtoken';
```
Comment (Japhethca) [inline src/routes/utils/helpers.js]: we can delete this file

**Review by Japhethca — CHANGES_REQUESTED** [review]
just few changes to make.

**Review by Japhethca — APPROVED** [review]

**Pull request merged**